src/librustc_session/utils.rs

```diff
@@ -11,17 +11,21 @@ impl Session {
 }
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
-pub enum NativeLibraryKind {
-    /// native static library (.a archive)
-    NativeStatic,
-    /// native static library, which doesn't get bundled into .rlibs
-    NativeStaticNobundle,
-    /// macOS-specific
-    NativeFramework,
-    /// Windows dynamic library without import library.
-    NativeRawDylib,
-    /// default way to specify a dynamic library
-    NativeUnknown,
+pub enum NativeLibKind {
+    /// Static library (`libfoo.a` or `foo.lib` on Windows) included when linking a final binary,
+    /// but not when archiving an rlib.
+    StaticNoBundle,
+    /// Static library (`libfoo.a` or `foo.lib` on Windows) included when linking a final binary,
+    /// but also included when archiving an rlib.
+    StaticBundle,
+    /// Dynamic library (`libfoo.so`) or an import library (`foo.lib`) on Windows.
+    Dylib,
+    /// Windows dynamic library (`foo.dll`) without a corresponding import library (`foo.lib`).
```

Comment:
`foo.lib` naming is only correct for MSVC. MinGW uses UNIX like naming for static libs `libfoo.a`, import libs can have various names and extensions but Rust sticks to the [preferred](https://github.com/bminor/binutils-gdb/blob/f4a220077b03af3a1f905b7dc6dc84c0a06d582f/ld/emultempl/pe.em#L2220-L2221) `libfoo.dll.a` format.

Reply:
I'll probably just remove the examples instead of covering all the possible extensions.

Reply:
Or better add more "for example"s.